### PR TITLE
udev/rules: add by-path and by-ibdev links to infiniband verbs

### PR DESCRIPTION
**Pull request by lkundrak**
The uverbs devices are sequentially numbered and are not guarranteed to stay stable across reboot.

At least one good person was disappointed by this, because they couldn't find their device: https://bugzilla.redhat.com/show_bug.cgi?id=2036515

Let's add a few helpful links.

### CODE REVIEW

File: rules.d/60-infiniband.rules [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# do not edit this file, it will be overwritten on update
+
+ACTION!="remove", SUBSYSTEM=="infiniband_verbs", IMPORT{builtin}="path_id"
+
+ENV{ID_PATH}=="?*", KERNEL=="uverbs*", SYMLINK+="infiniband/by-path/$env{ID_PATH}"
+ENV{ID_PATH}=="?*", KERNEL=="uverbs*", SYMLINK+="infiniband/by-ibdev/uverbs-$attr{ibdev}"
```
Comment (yuwata) [inline rules.d/60-infiniband.rules]: Please add a generic guard for "remove" action and non-infiniband devices at the beginning, something like the following:
```
ACTION=="remove", GOTO="infiniband_end"
SUBSYSTEM!="infiniband_verbs", GOTO="infiniband_end"
KERNEL!="uverbs*", GOTO="infiniband_end"
```

**Review by yuwata — CHANGES_REQUESTED** [review]

File: rules.d/60-infiniband.rules (hunk 1 above)
Comment (yuwata) [inline rules.d/60-infiniband.rules]: About the last line, maybe `ATTR{ibdev}=="?*"` instead of `ENV{ID_PATH}=="?*"`?

Comment (lkundrak) [reply rules.d/60-infiniband.rules]: Done.

Comment (lkundrak) [reply rules.d/60-infiniband.rules]: Done.

**Comment by lkundrak** [comment]
Thank you for your review & suggestions.

I've now updated the branch.

**Review by yuwata — APPROVED** [review]

**Comment by yuwata** [comment]
Looks (at least superficially) good to me.

But, let me confirm one point.
Why does systemd need to provide the .rules file, instead of other packages for infiniband, e.g. ibutils.rpm or so?

**Comment by yuwata** [comment]
> > Looks (at least superficially) good to me.
> > But, let me confirm one point. Why does systemd need to provide the .rules file, instead of other packages for infiniband, e.g. ibutils.rpm or so?
> 
> I think this is better to have it here, since it makes the behaviour consistent, no matter if you have some external package installed or not. The rules don't invoke anything external, all that is needed is the `path_id` builtin.

Yeah, I agree. The question in my previous comment does _not_ intend that the rule should be managed by another package, it's a pure question or confirmation.

**Pull request merged**